### PR TITLE
Cleaning up error/logging/panic handling in different places

### PR DESCRIPTION
**Pull request by alexanderConstantinescu**
Hi

This is a bit of a miscellaneous PR. It contains fixes for:

* Panics 
* Logging 
* Errors

which could be refactored as to provide for better error handling.

It also contains some changes for line indentation which looked really awkward.

### CODE REVIEW

File: go-controller/cmd/ovnkube/ovnkube.go [hunk 1]
```diff
@@ -194,51 +192,48 @@ func runOvnKube(ctx *cli.Context) error {
...
+		clusterController := ovncluster.NewClusterController(clientset, factory)
+		if err := clusterController.StartClusterNode(node); err != nil {
+			return err
+		}
 	}
 
-	return fmt.Errorf("need to run ovnkube in either master and/or node mode")
+	// run forever
+	select {}
+
```
Comment (dcbw) [inline go-controller/cmd/ovnkube/ovnkube.go]: unecessary whitespace addition

File: go-controller/cmd/ovnkube/ovnkube.go [hunk 2]
```diff
@@ -158,34 +156,34 @@ func setupPIDFile(pidfile string) error {
...
-		if err != nil {
+		if err := setupPIDFile(pidfile); err != nil {
 			return err
 		}
 	}
+
 	exec := kexec.New()
-	_, err := config.InitConfig(ctx, exec, nil)
-	if err != nil {
+
```
Comment (dcbw) [inline go-controller/cmd/ovnkube/ovnkube.go]: unecessary whitespace addition

File: go-controller/cmd/ovnkube/ovnkube.go [hunk 3]
```diff
@@ -158,34 +156,34 @@ func setupPIDFile(pidfile string) error {
 
 func runOvnKube(ctx *cli.Context) error {
 	pidfile := ctx.String("pidfile")
+
```
Comment (dcbw) [inline go-controller/cmd/ovnkube/ovnkube.go]: unecessary whitespace addition

File: go-controller/pkg/cluster/node.go [hunk 4]
```diff
@@ -200,14 +178,16 @@ func validateOVNConfigEndpoint(ep *kapi.Endpoints) bool {
 
 }
 
-func updateOVNConfig(ep *kapi.Endpoints) {
+func updateOVNConfig(ep *kapi.Endpoints) error {
+
```
Comment (dcbw) [inline go-controller/pkg/cluster/node.go]: unecessary whitespace addition

**Comment by dcbw** [comment]
Other than the whitespace and the travis failures, LGTM.

File: go-controller/pkg/cluster/node.go [hunk 5]
```diff
@@ -118,13 +103,9 @@ func (cluster *OvnClusterController) StartClusterNode(name string) error {
...
 	}
 
-	connected, err = isOVNControllerReady(name)
-	if err != nil {
+	if _, err = isOVNControllerReady(name); err != nil {
 		return err
 	}
-	if !connected {
-		return nil
-	}
```
Comment (girishmg) [inline go-controller/pkg/cluster/node.go]: @alexanderConstantinescu Not sure why you removed L125-127?

`isOVNControllerReady()` can return (false, nil). In that case we need to fail.

 Looks like the current code was incorrect in doing `return nil`.

Comment (alexanderConstantinescu) [reply go-controller/pkg/cluster/node.go]: Correct me if I am wrong, but I think the current function `isOVNControllerReady` can only return `(true, nil)` (L66) or `(false, error)` (L32, L50 , L64) in the current version. And if those two are the only two return conditions, then `!connected` becomes redundant.

Comment (alexanderConstantinescu) [reply go-controller/pkg/cluster/node.go]: If I am wrong however then we should still fix the return value for `!connected`. Returning `nil` means that we don't fail and thus do not force K8s to reboot the container. We thus don't fail but neither do we proceed to set up networking, we end up in "limbo"

Comment (alexanderConstantinescu) [reply go-controller/pkg/cluster/node.go]: Correct me if I am wrong, but `isOVNControllerReady` can only return `(true, nil)` (L66) or `(false, error)` (L32, L50, L64). If those are our only two possible conditions then that makes `!connected` redundant.

Comment (girishmg) [reply go-controller/pkg/cluster/node.go]: @alexanderConstantinescu you are right. I was thinking of something else, sorry about that.

Not this PR. Looks like `isOVNControllerReady()` should just return `error` and we should remove the `bool`.

**Pull request merged**